pelican/tests/test_paginator.py

```diff
@@ -76,3 +76,29 @@ def test_custom_pagination_pattern(self):
         page2 = paginator.page(2)
         self.assertEqual(page2.save_as, 'blog/2/index.html')
         self.assertEqual(page2.url, '//blog.my.site/2/')
+
+    def test_custom_pagination_pattern_last_page(self):
+        from pelican.paginator import PaginationRule
+        settings = get_settings()
+        settings['PAGINATION_PATTERNS'] = [PaginationRule(*r) for r in [
+            (1, '/{url}1/', '{base_name}/1/index.html'),
+            (2, '/{url}{number}/', '{base_name}/{number}/index.html'),
+            (-1, '/{url}', '{base_name}/index.html'),
+        ]]
+
+        self.page_kwargs['metadata']['author'] = Author('Blogger', settings)
+        object_list = [Article(**self.page_kwargs),
+                       Article(**self.page_kwargs),
+                       Article(**self.page_kwargs)]
+        paginator = Paginator('blog/index.html', '//blog.my.site/',
+                              object_list, settings, 1)
+        # The URL *has to* stay absolute (with // in the front), so verify that
+        page1 = paginator.page(1)
+        self.assertEqual(page1.save_as, 'blog/1/index.html')
+        self.assertEqual(page1.url, '//blog.my.site/1/')
+        page2 = paginator.page(2)
+        self.assertEqual(page2.save_as, 'blog/2/index.html')
+        self.assertEqual(page2.url, '//blog.my.site/2/')
+        page2 = paginator.page(3)
```

Comment:
I think this variable should be `page3`. 😛

Reply:
Fixed!